### DISCUSSION TITLE
I've fixed a lot missing types for hexo please review my repo

### DISCUSSION THREAD

**Discussion by dimaslanjaka** [hexojs/hexo#5199] [Show and tell]
i have fixed some missing typescript types for hexo, hexo-log, and some dependencies for hexo.

i love this library, 

please to review my improvisation at https://github.com/dimaslanjaka/hexo/tree/monorepo-v7

**Comment by yoshinorin** [comment]
@dimaslanjaka 

Thank you :)

However, would you please submit a PR to each repository for fixes types or dependencies? We will take a time for review, but we will do it as much as possible.

Thank you again.

**Reply by dimaslanjaka** [reply]
Okay ^_^

**Reply by dimaslanjaka** [reply]
please review https://github.com/hexojs/hexo-log/pull/148

cant log non-string parameter in typescript with `hexo.log` caused by typw `ConsoleArgs` is `string[]`

example
```ts
hexo.log.i('logname', { date: new Date() });
hexo.log.d('logname', ['foo', 'bar']);
```

above codes error like 

```log
Type 'Array' is not assignable to type 'string'. ts(2345)
```
